### ISSUE TITLE
Exact development branch/commit should be visible

### ISSUE THREAD

**Issue by jameshiebert** [pacificclimate/pdp#48]
In several places (e.g. in the [documentation](https://github.com/pacificclimate/pdp/blob/master/doc/source/conf.py#L55) and on the [web page templates](https://github.com/pacificclimate/pdp/blob/master/pdp/templates/map.html#L65)), we show a package "version number". However, in most cases this version number is only the latest child release number, since it is created from [manually updating a release number in `setup.py`](https://github.com/pacificclimate/pdp/blob/master/setup.py#L33). In most cases, the state version is kind of a lie.

There are many instances in which one would want to see the _exact_ commit hash or branch. For example:
- Reviewing changes in a development branch,
- Reviewing a deployed demo of something new feature that you want to show others before releasing it
- Reviewing documentation before releasing
- Running parallel versions of the software (e.g. a stable release and a dev release)

It would be best if these could be generated programmatically. Ideas?

``` bash
$ git rev-parse --short HEAD
e140dc4
$ git symbolic-ref HEAD | sed -e 's/refs\/heads\///'
feature/dockerize
```

I'm not sure when this should actually happen... if it's done at package run time, this doesn't make sense because it's not running from a git repo. Maybe best to use [`GitPython`](https://pypi.python.org/pypi/GitPython/python-git) and put this in the setup.py script somewhere?

**Comment by CBMasri** [comment]
Should the exact dev branch/commit replace the current `version` entirely? Or would it be better to include it as a new variable and add that to the docs as well (eg. `${version}: ${commitish}`)?

**Comment by jameshiebert** [comment]
I was just noticing this weekend how [readthedocs](https://readthedocs.org/) does it, which I think works pretty well. They host different "versions" of each packages docs (i.e. one from each branch), but down at the footer of each page, it also has the commit hash. See [here](http://pydap.readthedocs.io/en/develop/) for an example. It shows that it's on version "develop", but also that it was built from commit 26deab29.

In short, I think that your latter proposal would be best.
